tests/conftest.py

```diff
@@ -341,3 +343,34 @@ def teardown_subscribe_to_cdn_dogfood():
 def beta_env_setup(monkeypatch):
     # This fixture is used to set FOREMAN_MAINTAIN_USE_BETA env_var
     monkeypatch.setenv('FOREMAN_MAINTAIN_USE_BETA', '1')
+
+
+@pytest.fixture(scope='function')
+def setup_check_epel_repository(request, ansible_module):
+    testname = request.node.name  # To find name of test currently running.
+    setup = ansible_module.yum(
+        name=epel_repo,
+        state='present')
+    assert setup.values()[0]["rc"] == 0
+    if testname == "test_positive_check_epel_repository_with_invalid_repo":
```

Comment:
why it is needed? handle it at test level, else it may create lot of if else going forward and also make difficult to read the code

Reply:
Makes sense, updated PR.